src/SegmentedControl/SegmentedControl.jsx

```diff
@@ -22,19 +46,23 @@ export class SegmentedControl extends React.Component {
       return;
     }
 
-    this.setState({selected: value});
+    // Update internal state only if being used as an uncontrolled component.
+    if (!this.props.value) {
+      this.setState({selected: value});
+    }
```

Comment:
True - this conditional isn't really necessary, come to think of it. The internal need not depend on the value override. That override can just happen at render time.
Reverted it to try to simplify things a bit.